lib/include/openamp/rpmsg_virtio.h

```diff
@@ -145,6 +164,39 @@ int rpmsg_init_vdev(struct rpmsg_virtio_device *rvdev,
 		    struct metal_io_region *shm_io,
 		    struct rpmsg_virtio_shm_pool *shpool);
 
+/**
+ * rpmsg_init_vdev_with_config - initialize rpmsg virtio device with config
+ * Master side:
+ * Initialize RPMsg virtio queues and shared buffers, the address of shm can be
+ * ANY. In this case, function will get shared memory from system shared memory
+ * pools. If the vdev has RPMsg name service feature, this API will create an
```

Comment:
Minor English fixups:
s/has RPMsg name service/has the RPMsg name service/
s/an$/a/